### ISSUE TITLE
Validation split takes the first N documents (sorted, unshuffled) instead of a random sample

### ISSUE THREAD

**Issue by ParthibanS2810** [yazhi-lem/adhan#44]
## Summary
scripts/prepare_slm_corpus.py splits train/val like this:

    n_val = max(1, int(len(docs) * args.val_frac))
    val_docs = docs[:n_val]
    train_docs = docs[n_val:]

`docs` comes from `corpus.read_corpus()`, which walks source files in
sorted order and yields documents file-by-file
(src/adhan_slm/data/corpus.py:94). There's no shuffle anywhere before
the split. So val_docs is whatever happens to be alphabetically first,
not a representative sample.

## Reproduction
Built a synthetic 4-source corpus (250 docs each: sangam, wikipedia,
reddit, twitter). With val_frac=0.02:

    val  : Counter({'சங்கம்': 20})
    train: Counter({'விக்கிப்பீடியா': 250, 'ரெடிட்': 250,
                    'ட்விட்டர்': 250, 'சங்கம்': 230})

Validation set is 100% a single source.

## Impact
- val_loss / val_ppl (tracked in MLflow, used for best-checkpoint
  selection in train_jax.py) measures one register while training
  spans multiple sources — not a meaningful signal.
- The tokenizer is trained on train_docs only. Any akshara unique to
  the val-only source becomes <unk>, inflating val loss for a reason
  unrelated to model quality.
- --limit has the same problem: it truncates to the first N docs
  before any split, so debug/smoke runs also see one source.

## Fix
Shuffle documents with a fixed, logged seed before splitting (or
stratify by source file) in prepare_slm_corpus.py.

## Acceptance criteria
- [ ] val_docs / train_docs is a randomized (or stratified) split, not
      a raw prefix slice
- [ ] Split seed is recorded in the datasheet for reproducibility
- [ ] Re-running the reproduction above shows a mixed-source val set
